tests/test_integrated_channels/test_degreed/test_exporters/test_course_metadata.py

```diff
@@ -0,0 +1,113 @@
+        """
+        exporter = DegreedCourseExporter('fake-user', self.config)
+        # Add a course marked for deletion by giving it just the single key; the course run ID.
+        exporter.courses.append({'contentId': 'course-id-for-deletion'})
+        data = exporter.export()
+        json_dump1, method1 = (
+            # JSON Dump 1
+            json.dumps({
+                'courses': [{'contentId': 'course-id-for-deletion'}],
+                'orgCode': self.config.degreed_company_id,
+                'providerCode': 'EDX',
+            }, sort_keys=True).encode('utf-8'),
+
+            # Method 1
+            'DELETE'
+        )
+        json_dump2, method2 = (
+            # JSON Dump 2
+            json.dumps({
+                'courses': [{
+                    'authors': [],
+                    'categoryTags': [],
+                    'contentId': 'course-v1:edX+DemoX+Demo_Course_1',
+                    'costType': 'Paid',
+                    'description': 'edX Demonstration Course',
+                    'difficulty': '',
+                    'duration': 0,
+                    'format': 'Instructor',
+                    'imageUrl': '',
+                    'institution': '',
+                    'language': 'en',
+                    'publishDate': '2013-02-05',
+                    'title': 'edX Demonstration Course',
+                    'url': 'http://localhost:8000/enterprise/' + str(self.enterprise_customer.uuid) +
+                           '/course/course-v1:edX+DemoX+Demo_Course_1/enroll/',
+                    'videoUrl': '',
+                }],
+                'orgCode': self.config.degreed_company_id,
+                'providerCode': 'EDX',
+            }, sort_keys=True).encode('utf-8'),
+
+            # Method 2
+            'POST'
+        )
+
+        assert next(data) == (json_dump1, method1)
+        assert next(data) == (json_dump2, method2)
+
+    @responses.activate
+    def test_transform_unenrollable_course(self):
+        unenrollable_course_run = {
+            'key': 'course-key',
+            "enrollment_start": "2999-10-13T13:11:03Z"
+        }
+        exporter = DegreedCourseExporter('fake-user', self.config)
+        assert exporter.transform(unenrollable_course_run) == {'contentId': 'course-key'}
+
+    @responses.activate
+    def test_transform_description_returns_full_description(self):
+        course_run = {'full_description': 'a' * (DegreedCourseExporter.LONG_STRING_LIMIT - 1)}
+        exporter = DegreedCourseExporter('fake-user', self.config)
+        assert exporter.transform_description(course_run) == course_run['full_description']
+
+    @responses.activate
+    def test_transform_description_returns_empty_str(self):
+        course_run = {'full_description': 'a' * (DegreedCourseExporter.LONG_STRING_LIMIT + 1)}
+        exporter = DegreedCourseExporter('fake-user', self.config)
+        assert exporter.transform_description(course_run) == ''
```

Comment:
@UmanShahzad Let's add docstrings to the three tests above.

Reply:
:+1: